specs/_features/ePBS/validator.md

```diff
@@ -0,0 +1,250 @@
+
+```python
+def get_ptc_assignment(state: BeaconState,
+                             epoch: Epoch,
+                             validator_index: ValidatorIndex
+                             ) -> Optional[Tuple[Sequence[ValidatorIndex], Slot]]:
+    """
+    Return the ptc committee assignment in the ``slot`` for ``validator_index``.
+    ``assignment`` returned is a tuple of the following form:
+        * ``assignment[0]`` is the list of validators in the ptc
+        * ``assignment[1]`` is the slot at which the ptc is assigned
+    Return None if no assignment.
+    """
+    next_epoch = Epoch(get_current_epoch(state) + 1)
+    assert epoch <= next_epoch
+
+    start_slot = compute_start_slot_at_epoch(epoch)
+    committee_count_per_slot = get_committee_count_per_slot(state, epoch)
+    for slot in range(start_slot, start_slot + SLOTS_PER_EPOCH):
+        for index in range(committee_count_per_slot):
+            committee = get_ptc(state, Slot(slot))
+            if validator_index in committee:
+                return committee, Slot(slot)
+    return None
+```
+
+### Lookahead
+
+The beacon chain shufflings are designed to provide a minimum of 1 epoch lookahead
+on the validator's upcoming ptc assignments for attesting dictated by the shuffling and slot.
+
+[New in ePBS]
+
+`get_ptc_assignment` should be called at the start of each epoch to get the assignment for the next epoch (`current_epoch + 1`).
+A validator should plan for future assignments by noting their assigned ptc committee
+slot and planning to participate in the ptc committee subnet.
+
+[Modified in MaxEB]
+
+a validator should:
+* Find peers of the pubsub topic `beacon_attestation_{subnet_id}`.
+    * If the validator is assigned to be an aggregator for the slot (see `is_aggregator()`)[Modified in MaxEB], then subscribe to the topic.
+
+### Inclusion list proposal
+
+ePBS introduces forward inclusion list. The detail design is described in this [post](https://ethresear.ch/t/no-free-lunch-a-new-inclusion-list-design/16389)
+Proposer must construct and broadcast `InclusionList` alongside `SignedBeaconBlock`.
+- Proposer for slot `N` submits `SignedBeaconBlock` and in parallel submits `InclusionList` to be included at the beginning of slot `N+1` builder.
+- Within `InclusionList`, `Transactions` are list of transactions that the proposer wants to include at the beginning of slot `N+1` builder.
+- Within `inclusionList`, `Summaries` are lists consisting on addresses sending those transactions and their gas limits. The summaries are signed by the proposer `N`.
+- Proposer may  send many of these pairs that aren't committed to its beacon block so no double proposing slashing is involved.
+
+#### Constructing the inclusion list
+
+To obtain an inclusion list, a block proposer building a block on top of a `state` must take the following actions:
+
+1. Retrieve inclusion list from execution layer by calling `get_execution_inclusion_list`.
```

Comment:
I think proposer should also check if previous slot's payload was empty or not. If it was, the proposer shouldn't send a new inclusion list.

Reply:
Okay I think this info has been added into the fork choice section by Potuz in a code snippet so not sure if we need it here.